vscode-dotty/src/extension.ts

```diff
@@ -43,47 +68,139 @@ export function activate(context: ExtensionContext) {
   })
 
   if (process.env['DLS_DEV_MODE']) {
-    const portFile = `${vscode.workspace.rootPath}/.dotty-ide-dev-port`
+    const portFile = path.join(workspaceRoot, ".dotty-ide-dev-port")
     fs.readFile(portFile, (err, port) => {
       if (err) {
         outputChannel.appendLine(`Unable to parse ${portFile}`)
         throw err
       }
 
       run({
-        module: context.asAbsolutePath('out/src/passthrough-server.js'),
+        module: context.asAbsolutePath(path.join("out", "src", "passthrough-server.js")),
         args: [ port.toString() ]
       }, false)
     })
 
   } else {
-    // Check whether `.dotty-ide-artifact` exists. If it does, start the language server,
-    // otherwise, try propose to start it if there's no build.sbt
-    if (fs.existsSync(languageServerArtifactFile)) {
-      runLanguageServer(coursierPath, languageServerArtifactFile)
-    } else if (!fs.existsSync(disableDottyIDEFile) && !fs.existsSync(buildSbtFile)) {
-      vscode.window.showInformationMessage(
-          "This looks like an unconfigured Scala project. Would you like to start the Dotty IDE?",
-          "Yes", "No"
+    let configuredProject: Thenable<void> = Promise.resolve()
+    if (isUnconfiguredProject()) {
+      configuredProject = vscode.window.showInformationMessage(
+        "This looks like an unconfigured Scala project. Would you like to start the Dotty IDE?",
+        "Yes", "No"
       ).then(choice => {
-        if (choice == "Yes") {
-          fs.readFile(languageServerDefaultConfigFile, (err, data) => {
-            if (err) throw err
-            else {
-              const languageServerScalaVersion = data.toString().trim()
-              fetchAndConfigure(coursierPath, sbtArtifact, languageServerScalaVersion, dottyPluginSbtFile).then(() => {
-                runLanguageServer(coursierPath, languageServerArtifactFile)
-              })
-            }
-          })
-        } else {
+        if (choice === "Yes") {
+          bootstrapSbtProject(buildSbtFileSource, dottyPluginSbtFileSource)
+          return Promise.resolve()
+        } else if (choice === "No") {
           fs.appendFile(disableDottyIDEFile, "", _ => {})
+          return Promise.reject()
         }
       })
     }
+
+    configuredProject
+      .then(_ => connectToSbt(coursierPath))
```

Comment:
Since starting sbt and running configureIDE might be very slow, I suggest we do not do it if the project has already been configured once (i.e., .dotty-ide-artifact, .dotty-ide.json are present). In the future, the Dotty Language Server will decide itself if it should query the build tool for configuration options.